### ISSUE TITLE
Run PR workflow only when code changes

### ISSUE THREAD

**Issue by NDStrahilevitz** [aquasecurity/tracee#1608]
Currently, every PR, including ones that only change documentations or add docs, will trigger the PR workflow which checks for unit tests, integration and smoke tests. 
We can easily filter out those PRs from running the workflow through a path filter as documented [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths).

I'm opening this issue to ask if this should be done, and if so, which paths should be included and/or excluded.

**Comment by danielpacak** [comment]
I think it's a good idea, but implementing it with GitHub may be tricky. We have enabled branch protection rule that **Require status checks to pass before merging PR**. However, AFAIK branch protection rules cannot be skipped on the same paths you add to the GitHub Actions workflow config. Therefore, you'll end up skipped GitHub Actions workflow, but status check won't be updated. Only repository admins will be able to merge such PR.

Here's a screenshot from another project where I used exclude paths:

![exclude_and_checks_github](https://user-images.githubusercontent.com/1322923/160585145-be30bdef-6751-4ae4-93df-cb0260c88d53.png)

As you can see the **Squash and merge** button in disabled because the PR is somehow stuck in the *Waiting for status to be reported* state.

**Comment by danielpacak** [comment]
Correct. No matter how many workflows we have, with or without exclusion rules, they are not respected by branch protection rules that you configure at GitHub repository level. I think it's kind of the limitation of GitHub itself. Maybe we can dig into the documentation and find out official solution, but at the time I read it it wasn't possible.

**Comment by rafaeldtinoco** [comment]
BTW, this has also to include the "eBPF code change" will trigger new CO-RE tests (https://github.com/aquasecurity/tracee-test-kernels/tree/main/tester) being integrated into PR workflow.

**Comment by yanivagman** [comment]
This seems to be already in place, isn't it?
https://github.com/aquasecurity/tracee/blob/main/.github/workflows/pr.yaml#L19
@geyslan @NDStrahilevitz Can we close this?
